### DISCUSSION TITLE
Security handlers

### DISCUSSION THREAD

**Discussion by jeffreybradley1963** [oas-tools/oas-tools.github.io#18] [Q&A]
Are there any examples of how to decrypt/decode the token that is passed into the BasicAuth and CookieAuth handlers?  I've tried cookieParser, but I cannot it get it to work.

**Comment by jeffreybradley1963** [comment]
I figured out how to get my cookie value.  I had to take the token, and run it through decodeURIComponent before running it through the cookieParser.signedCookie function.  This gives me my session id, but since I don't have the request object, I am not sure how to ask express sessions to give me information in the session so I can validate it. I want to make sure the session still exists as well as pull some information that I stick into the session to let me know what the user can do.

**Comment by jeffreybradley1963** [comment]
I think I solved it.  I took care of extracting the tokens and then I added an external module that fires before the routing takes place.  It gets the request and response object and now I can handle any session management.

**Comment by alesancor1** [comment]
Have you checked the [security section](https://oas-tools.github.io/docs/features/security) in the docs? In order to get the token from the Cookie you just need to declare a SecurityScheme with `type: apiKey` and `in: cookie`, then declare a securityHandler function (as explained at the bottom of the docs page) that receives the token.

If something is unclear in documentation please don't be afraid to leave an issue or PR. I'm glad you found a workaround using an external module anyway :)
